src/Language/PureScript/Sugar/TypeClasses.hs

```diff
@@ -79,12 +80,22 @@ desugarModule
   -> Module
   -> Desugar m Module
 desugarModule syns kinds (Module ss coms name decls (Just exps)) = do
-  let (classDecls, restDecls) = partition isTypeClassDecl decls
+  decls' <- parU decls desugarInstName
+  let (classDecls, restDecls) = partition isTypeClassDecl decls'
```

Comment:
The `partition` line currently iterates through all declarations. By adding `parU decls desugarInstName`, declarations will be iterated through a second time. Ideally, we would only iterate through the list once and get the results of both.

Reply:
Why `parU`? `desugarInstName` can't error; isn't this just a `traverse`?

Reply:
Ah, yeah it is.

Also, if we used one fold to produce the results of both, I believe we'd need to reverse the resulting `restDecls` list, which would iterate through the list a second time and might defeat the point of this approach.

Reply:
And then I suppose if you're concerned about extra iterations, could you just apply this to `restDecls` instead of all of `decls`?

Reply:
Yeah, good point.